modules/notification/base/null.go

```diff
@@ -54,6 +54,10 @@ func (*NullNotifier) NotifyPullRequestSynchronized(doer *models.User, pr *models
 func (*NullNotifier) NotifyPullRequestChangeTargetBranch(doer *models.User, pr *models.PullRequest, oldBranch string) {
 }
 
+// NotifyPullRequestPushCommits notifies when push commits to pull request's head branch
+func (*NullNotifier) NotifyPullRequestPushCommits(doer *models.User, pr *models.PullRequest, comment *models.Comment) {
```

Comment:
I thought we already have a type for that?

Reply:
(Webhooks can be triggerd by push, and as far  as i can remember they use the same code...)

Reply:
Do you means this  function.``NotifyPushCommits(pusher *models.User, repo *models.Repository, refName, oldCommitID, newCommitID string, commits *repository.PushCommits)``, I didn't use it because  it not have pull request message . if use it, will need to do other thing to get pr message. so I think use a new notify type is necessary. How about you? Thanks

Reply:
Hmm, in this view the new type is ok

Reply:
By the way, maybe should also check if it is force-push in here , if true should do other thing or just skip add feed actions. because won't get right commit list ....
https://github.com/go-gitea/gitea/blob/e008b68c9c312681d4098f6f3b8f208479f4f43a/modules/repofiles/update.go#L602-L607